src/ui/auth/withPermissions.js

```diff
@@ -1,59 +1,95 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { compose } from 'redux';
 import { Spinner } from 'patternfly-react';
 import { PermissionCheck, routeConverter } from '@entando/utils';
 import { NoAccessPage } from '@entando/pages';
 import { withRouter } from 'react-router-dom';
 import { connect } from 'react-redux';
 import { logoutUser } from '@entando/apimanager';
 import { ROUTE_DASHBOARD } from 'app-init/router';
-import { ADMINISTRATION_AREA_PERMISSION } from 'state/permissions/const';
+import { hasAccess } from '@entando/utils/dist/permissions';
+import {
+  ADMINISTRATION_AREA_PERMISSION,
+  ROLE_SUPERUSER,
+} from 'state/permissions/const';
 import { getLoggedUserPermissions } from 'state/permissions/selectors';
 
+const injectPermissionValues = WrappedComponent => (props) => {
+  // eslint-disable-next-line react/prop-types
+  const { userPermissions, ...otherProps } = props;
+
+  const isSuperuser = hasAccess(ROLE_SUPERUSER, userPermissions);
```

Comment:
please add `userPermissions || []` instead of `userPermissions`. Because, when logging in, we have default value of null for `userPermissions` in reducer and `hasAccess` gives type error. `null` is because of spinner while permissions are being fetched. It is complex, because Keycloak differently handles permissions and legacy does it differently. `null` was only cure for handling all scenarios.